### PR TITLE
Created tvOS Example Pack

### LINKED ISSUE #138: Add tvOS example project
Depends on #118 and #137.

Let's get a basic tvOS example setup.

Let's put this in a top-level dir, `Examples-tvOS/`

### PR DESCRIPTION
**Pull request by Sherlouk**
## Changes in this pull request
- Created a new workspace for tvOS examples w/ Cocoapods & dependency on IGListKit
- Created a demo chooser across the same lines as the iOS equivalent
- Duplicated the 'Nested Adapter' example from iOS pack to the tvOS example
  - Uses native navigation to get around, created focus/active states for cells
- Updated `travis.yml` to optionally build tvOS example pack

Note: Not added any imagery/icons/etc due to the much larger & different sizes required - If they're desired then someone with skills will need to provide them!

Closes #138 
Relates to #137 
## Pull request checklist
- [x] All tests pass. Demo project builds and runs.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by Sherlouk** [comment]
Having a few issues with the travis change, wasn't working locally when I tried but a few retrys later it worked locally.

Would appreciate some help with that!

**Comment by rnystrom** [comment]
@Sherlouk It looks like Travis is timing out from a build issue. You should be able to run this locally:

```
$ xcodebuild build -workspace Examples-tvOS/IGListKitExamples.xcworkspace -scheme IGListKitExamples -sdk "$TVOS_SDK" -destination "OS=9.0,name=Apple TV 1080p" ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO -verbose
```

That's the command its timing out on (+ `-verbose`): https://travis-ci.org/Instagram/IGListKit/jobs/171743818

File: .travis.yml [hunk 1]
```diff
@@ -18,32 +18,37 @@ env:
...
 
 script:
 - if [ $POD_LINT == "YES" ]; then
       bundle exec pod lib lint;
   fi
 
 
-- if [ $BUILD_EXAMPLE == "YES" ]; then
+- if [ $BUILD_IOS_EXAMPLE == "YES" ]; then
       xcodebuild build -workspace Example/IGListKitExamples.xcworkspace -scheme IGListKitExamples -sdk "$SDK" -destination "$DESTINATION" ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO | bundle exec xcpretty -c;
```
Comment (jessesquires) [inline .travis.yml]: Let's revert these changes. Instead, inside of this `if` we can check the value of `$SDK`

File: Examples-tvOS/Pods/Manifest.lock [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+EXTERNAL SOURCES:
+  IGListKit:
+    :path: ../IGListKit.podspec
+
+SPEC CHECKSUMS:
+  IGListKit: a534a917b574ec56697390ef182f125ccff2145b
+
+PODFILE CHECKSUM: 4bdfb42d1e7b2b121bf8c83bc300d7d77aa0fc3a
+
+COCOAPODS: 1.0.1
```
Comment (jessesquires) [inline Examples-tvOS/Pods/Manifest.lock]: let's use `1.1.1`

**Comment by Sherlouk** [comment]
With regards to Travis, definitely not my strong point but will try and get it working (will make the suggested changes first!)

Definitely happy with making things final and updating the cocoapods version.

Are we satisfied with the demo I chose to do/the chooser I implemented?

I'm actually with you @rnystrom I've actually never done tvOS hence why I wanted to do this to try and learn something new :) (Had fun learning how to focus cells and doing all that!)

**Comment by Sherlouk** [comment]
Travis hasn't run yet, but I haven't changed the build command so I doubt the result will change.

When I ran the build command locally it didn't work at first, it stalled because I had a different build of xcode open than it wanted to use. It was forcing itself to use xcode 8 but I had 8.beta2 open. I closed it, and reran the command and it worked perfectly.

I've updated my local version of cocoapods to 1.1.1, made classes final and made the travis file a bit cleaner as per requests.

**Comment by Sherlouk** [comment]
As much as I'd love to say I fixed it -- I broke the build shell script. 🙈 

Working on it

**Comment by Sherlouk** [comment]
Failed in same place, but no closer to finding the actual issue.

As I mentioned previously working fine when running locally, so let's just ship my machine!

But in all seriousness, any ideas @rnystrom ?

File: .travis.yml [hunk 3]
```diff
@@ -30,20 +30,25 @@ env:
...
+- if [ $BUILD_EXAMPLE == "YES" ] && [ $SDK == $IOS_SDK ]; then
       xcodebuild build -workspace Example/IGListKitExamples.xcworkspace -scheme IGListKitExamples -sdk "$SDK" -destination "$DESTINATION" ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO | bundle exec xcpretty -c;
   fi
 
 
+- if [ $BUILD_EXAMPLE == "YES" ] && [ $SDK == $TVOS_SDK ]; then
+      xcodebuild build -workspace Examples-tvOS/IGListKitExamples.xcworkspace -scheme IGListKitExamples -sdk "$SDK" -destination "$DESTINATION" ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO | bundle exec xcpretty -c;
+  fi
+
+
```
Comment (jessesquires) [inline .travis.yml]: 👍

**Comment by jessesquires** [comment]
@Sherlouk -- are the tvOS example schemes shared?

**Comment by Sherlouk** [comment]
@jessesquires Yeah, obviously, who would forget to do that? Insane idea!

(oops)

**Comment by Sherlouk** [comment]
That was slightly embarrassing. 😓

**Comment by rnystrom** [comment]
@Sherlouk what's the reasoning behind another sample project vs adding this all to the existing one?

screen shot 2016-11-07 at 9 50 13 am

IMO I'd like to keep the top-level dir as trim as possible (which is becoming tough). Can we not add this to the existing sample project? Maybe I missed something?

**Comment by Sherlouk** [comment]
@rnystrom It originally was all in one project file, but redid it all in a fresh project file as per @jessesquires' comments (https://github.com/Instagram/IGListKit/issues/138#issue-186094900)

**Comment by Sherlouk** [comment]
@rnystrom Any update on this?

**Comment by Sherlouk** [comment]
@rnystrom @jessesquires 

Any note on what I need to do to move this forward?

**Comment by jessesquires** [comment]
@rnystrom -- Yeah, I think we should have a separate tvOS example project.

Let's keep directories how they are for this PR and follow up after. 

Here's what I propose:

```
|-- Examples/
    |-- Examples-iOS/
    |-- Examples-tvOS/
```

**Comment by jessesquires** [comment]
I think we're good to go here.

@rnystrom -- Any other review comments? 

@Sherlouk -- Sorry for the delay here! Looks like there's one conflict in `CHANGELOG.md`. Sorry about that. Should be easy to fix though.

Follow up task to clean up directories: #202

**Comment by Sherlouk** [comment]
@jessesquires Not a problem, was waiting for everybody to be happy before fixing it as other PRs kept making it conflict!

Should all work now 🙏

**Comment by rnystrom** [comment]
Let's land it! 🚀 🚀 🚀
